qdrant-landing/content/articles/vector-search-resource-optimization.md

```diff
@@ -0,0 +1,598 @@
+| **Plan for Scalability**        | Start with at least **2 shards per node** to allow room for future growth.                                        |
+| **Future-Proofing**             | Starting with around **12 shards** is a good rule of thumb. This setup allows your system to scale seamlessly from 1 to 12 nodes without requiring re-sharding. |
+| **Avoid Re-Sharding**           | Re-sharding can be complex and requires creating a new collection, so it's best to plan ahead to minimize disruptions. |
+
+Learn more about [**Sharding in Distributed Deployment**](/documentation/guides/distributed_deployment/)
+
+---
+
+## Query Optimization
+![qdrant](/articles_data/vector-search-resource-optimization/query.png)
+Improving vector database performance is critical when dealing with large datasets and complex queries. By leveraging techniques like **filtering**, **batch processing**, **reranking**, **rescoring**, and **oversampling**, so you can ensure fast response times and maintain efficiency even at scale.
+
+#### Filtering
+
+Filtering allows you to select only the required fields in your query results. By limiting the output size, you can significantly reduce response time and improve performance.
+
+The filterable vector index is Qdrant's solves pre and post-filtering problems by adding specialized links to the search graph. It aims to maintain the speed advantages of vector search while allowing for precise filtering, addressing the inefficiencies that can occur when applying filters after the vector search.
+
+**Example:**
+
+```python
+results = client.search(
+    collection_name="my_collection", 
+    query_vector=[0.1, 0.2, 0.3], 
+    limit=10, 
+    with_payload={"include": ["category"]}
+)
+```
+**Figure 7:** The filterable vector index adds specialized links to the search graph to speed up traversal.
+
+![filterable-vector-index](/articles_data/vector-search-resource-optimization/filterable-vector-index.png)
+
+[**Filterable vector index**](/documentation/concepts/indexing/): This technique builds additional links **(orange)** between leftover data points. The filtered points which stay behind are now traversible once again. Qdrant uses special category-based methods to connect these data points.
+
+---
+Read more about [**Filtering Docs**](/documentation/concepts/filtering/) and check out the [**Complete Filtering Guide**](/articles/vector-search-filtering/).
+
+---
+#### Batch Processing
+
+Batch processing consolidates multiple operations into a single execution cycle, reducing transaction overhead and enhancing throughput. It’s an effective strategy for both data insertion and query execution.
+
+<img src="/articles_data/vector-search-resource-optimization/batch-processing.png" alt="batch-processing" style="width: 75%;">
+
+**Batch Insertions**: Instead of inserting vectors individually, group them into larger batches to minimize the number of database transactions and the overhead of frequent writes.
```

Comment:
Too large batches are not ideal either. Let's call it medium-sized batches.
Also, there are no transactions in Qdrant